pkg/codegen/json/README.md

```diff
@@ -0,0 +1,98 @@
+# PCL to JSON (experimental)
+
+This directory contains a PCL to JSON code generator which converts a PCL program into a JSON format that represents the Abstract Syntax Tree (AST) of the program. This JSON can be used by external tools to perform analysis or conversion of PCL programs to target languages without needing to build a specialized PCL parser. The JSON format is not intended to be human readable, but rather to be easily parsed by external tools.
+
+## Usage
+Execute `pulumi convert` against a Pulumi YAML program in experimental mode:
+```
+PULUMI_EXPERIMENTAL=1 pulumi convert --language json --out json
+```
+
+Note that Pulumi already type checks the program before converting it to JSON. If the program contains errors, the conversion will fail.
+
+## Format
+
+A PCL program is converted into a JSON object with a field called `nodes` which is an array of the nodes of the program:
+```json
+{
+    "nodes": [...]
```

Comment:
This is throwing away all trivia information. I've been inclined to saying we probably _should_ do that because in general translating trivia into all languages is silly hard (how do you translate `attr = /* some comment */ value // another comment` to python for example?)
But we should probably have some way for code generators to maintain some comments when translating from PCL. 
Like if Resource had a description, but also each input and option. Complicates the model a fair bit but means you could at least maintain comments like:
```
// This is a bucket
resource "bucket" "aws:s3:Bucket" { ....
```

Reply:
Trivia can be added in a later iteration, we don't to have to throw it away. Just getting this into a usable shape was my priority